meinberlin/apps/dashboard/views.py

```diff
@@ -105,3 +115,165 @@ def validate_object_project(self):
 
     def validate_object_module(self):
         return True
+
+
+class AbstractProjectUserListView(a4dashboard_mixins.DashboardComponentMixin,
+                                  a4dashboard_mixins.DashboardBaseMixin,
+                                  a4dashboard_mixins.DashboardContextMixin,
+                                  generic.base.TemplateResponseMixin,
+                                  generic.edit.FormMixin,
+                                  generic.detail.SingleObjectMixin,
+                                  generic.edit.ProcessFormView):
+
+    form_class = forms.AddUsersFromEmailForm
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        if 'submit_action' in request.POST:
+            if request.POST['submit_action'] == 'remove_user':
+                pk = int(request.POST['user_pk'])
+                user = get_object_or_404(User, pk=pk)
+
+                if request.POST['submit_action'] == 'remove_user':
+                    related_users = getattr(self.object,
+                                            self.related_users_field)
+                    related_users.remove(user)
+                    messages.success(request, self.success_message_removal)
+
+            return redirect(self.get_success_url())
+        else:
+            return super().post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        if form.missing:
+            messages.error(
+                self.request,
+                _('Following emails are not registered: ') + ', '.join(
+                    form.missing)
+            )
+
+        if form.cleaned_data['add_users']:
+            users = form.cleaned_data['add_users']
+            related_users = getattr(self.object,
+                                    self.related_users_field)
+            related_users.add(*users)
+
+            messages.success(
+                self.request,
+                ungettext(self.success_message[0], self.success_message[1],
+                          len(users)).format(len(users))
+            )
+
+        return redirect(self.get_success_url())
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['label'] = self.add_user_field_label
+        return kwargs
+
+
+class AbstractProjectUserInviteListView(AbstractProjectUserListView):
```

Comment:
i would merge this with AbstractProjectUserListView as AbstractProjectUserListView is not used anywhere else

Reply:
I was thinking that this might be useful for some other feature later.